rfc/rfc_001_stamping-txs.md

```diff
@@ -0,0 +1,117 @@
+.                        .   |       .                        .     /-----> Tx-Hash (32 bytes)     .
+|                        |   |       |                        |     |     |                        |
++------------------------+   |       +------------------------+     |     +------------------------+
+                             |                                      |
+                             |                                      |
+                             |                                      |
+                             |          Tx                          |
+                             |         +-------------------+\       |
+                             \-------->| STAMP             ||       |
+                                       +-------------------+|       |
+                                       | PAYLOAD           ||       |
+                                       |                   ||-------/
+                                       |                   ||
+                                       |                   ||
+                                       +-------------------+|
+                                       | SIGNATURE         ||
+                                       +-------------------+/
+```
+
+### Where stamped transactions will be saved?
+
+Full nodes can have separate database for saving stamped transactions.
+This database will only be used for querying transaction. A node can choose to save all txs, part of them or nothing.
+Nodes can query transaction by hash from each other. Users also can save their own transactions in their own wallets.
+
+
+### Can a stamped transaction be lost?
+
+Probably yes and it's not a curse. Should we care about transaction happened 10 years ago and try to keep it in our DB?
+By decoupling transaction from the block, transactions will have their own way to survive.
+
+
+## Decision
+
+## Status
+
+## Consequences
+
+* Reducing block size.
+
+  Stamping transaction can reduce the block size dramatically. For keeping a stamped transaction inside a block we need to store only 32 bytes hash of transaction instead of whole transaction data. In the bitcoin, the average transaction size has increased from 450 bytes in Jan 2013 to almost 600 bytes in Oct 2015. In the below chart you can see how stamped transaction could reduce the size of bitcoin blockchian if they could had applied from the beginning:
+
+  ![Bitcoin average block size](img/btc-block-size.png)
+  ![Bitcoin blockchain size](img/btc-blockchain-size.png)
+
+* Easy light clint implementation for PoS
+
+  Implementing a light client for PoW is easy since having block header is fair enough to check validity of a transaction using merkle tree.
+  However in PoS implementing a light client is a big pain. By stamping transaction light client implementation will be easy. A light client doesn't need to store any transaction data.
```

Comment:
How did this change the lite client in PoS? This doesn't affect the hard of a PoS liteclient, authenticating a block header. Once you have that, the state query and/or tx query is very easy, just like in PoW.

Reply:
Good point. Maybe needs to change this part.
Maybe we don't need to have a lite client anymore. A lite client is simply a client without any TX-DB.